From 28328b6403d8f4f842feab08d0e43d66fdfe8726 Mon Sep 17 00:00:00 2001
From: "Gustavo A. R. Silva" <garsilva@embeddedor.com>
Date: Fri, 6 Mar 2020 14:37:20 -0800
Subject: [PATCH 08/19] datapath: conntrack: mark expected switch fall-through

Upstream commit:
    commit 279badc2a85be83e0187b8c566e3b476b76a87a2
    Author: Gustavo A. R. Silva <garsilva@embeddedor.com>
    Date:   Thu Oct 19 12:55:03 2017 -0500

    openvswitch: conntrack: mark expected switch fall-through

    In preparation to enabling -Wimplicit-fallthrough, mark switch cases
    where we are expecting to fall through.

    Notice that in this particular case I placed a "fall through" comment on
    its own line, which is what GCC is expecting to find.

    Signed-off-by: Gustavo A. R. Silva <garsilva@embeddedor.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Acked-by: Yi-Hung Wei <yihung.wei@gmail.com>
Signed-off-by: Greg Rose <gvrose8192@gmail.com>
Signed-off-by: Ben Pfaff <blp@ovn.org>
---
 datapath/conntrack.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/datapath/conntrack.c b/datapath/conntrack.c
index b7eb53f93..5b4d6cce0 100644
--- a/datapath/conntrack.c
+++ b/datapath/conntrack.c
@@ -844,6 +844,7 @@ static int ovs_ct_nat_execute(struct sk_buff *skb, struct nf_conn *ct,
 			}
 		}
 		/* Non-ICMP, fall thru to initialize if needed. */
+		/* fall through */
 	case IP_CT_NEW:
 		/* Seen it before?  This can happen for loopback, retrans,
 		 * or local packets.
-- 
2.17.1

